packages/data/src/viem.ts

```diff
@@ -0,0 +1,241 @@
+import { Chain, PublicClient, http, createPublicClient, getContract, GetContractReturnType } from "viem"
+import { arbitrum, arbitrumGoerli, polygon, goerli, sepolia, optimismGoerli } from "viem/chains"
+import SemaphoreABI from "./semaphoreABI"
+import { GroupResponse, ViemOptions } from "./types"
+import checkParameter from "./checkParameter"
+
+export default class SemaphoreViem {
+    private _client: PublicClient
+    private _options: ViemOptions
+    private _contract: GetContractReturnType<typeof SemaphoreABI, PublicClient>
+
+    /**
+     * Initializes the Viem object with an viem chain.
+     * @param network Chain object of viem.
+     * @param options Viem options.
+     */
+    constructor(network: Chain = sepolia, options: ViemOptions = {}) {
```

Comment:
It would be nice for devs to have a way to pass a valid chain without installing `vite` separately and importing `viem/chains`. The type here can be similar to `EthersNetwork` (i.e. strings), and the constructor could then use the right chain object based on that string. It would also be more consistent with `SemaphoreEthers`.

Reply:
I think the current implementation is better here, given its affinity with the Viem ecosystem.